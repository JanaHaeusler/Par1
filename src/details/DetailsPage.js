import PropTypes from 'prop-types'
import { useHistory } from 'react-router-dom'
import styled from 'styled-components/macro'
import { v4 as uuid } from 'uuid'
import { BackIconDark } from '../app/Icons/Icons'

DetailsPage.propTypes = {
    targetProfile: PropTypes.object.isRequired,
}

export default function DetailsPage({targetProfile}) {
    
    const history = useHistory()
    const {location, date, playersString, winner, shots} = targetProfile
    const players = playersString

    return (
        <GameDetailsCard>
            <GameKeyInfos> 
                <Location>{location}</Location>
                <Date>{date}</Date>
                <Players>
                    <h4>Player(s)</h4>
                    <span>{players}</span>
                </Players>
                <Winner>
                    <h4>Winner(s)</h4>
                    <span>{winner}</span>
                </Winner>
                <Shots>
                    <h4>Total Shots</h4>
                    <span>{shots}</span>
                </Shots>
            </GameKeyInfos>
            <GameScores>
                    <LegendHoles>
                        <span>Holes</span>
                        {new Array(18).fill().map((_, index) => {
                            const newId = uuid()
                            return <span key={newId}>{index + 1}</span>
                            })
                        }
                    </LegendHoles> 
                    <ScoresAllPlayers>
                        {targetProfile.playersArray?.map((player) => {
                            const newId = uuid()
                            const playerName = player     
                                return (
                                    <ScoresSinglePlayer key={newId}>
                                        <span>{playerName}</span>
                                        {new Array(18).fill().map((_, index) => {
                                            const newId = uuid()
                                            const fieldNumber = index + 1
                                            const holeName = 'hole' + fieldNumber
                                            return( 
                                                <SingleScore key={newId} >
                                                    {targetProfile.scores[playerName][holeName]}
                                                </SingleScore>
                                                )                         
                                            })
                                        }
                                    </ScoresSinglePlayer>
                                )
                        })}
                    </ScoresAllPlayers> 
                </GameScores> 
            <ButtonWrapper>
                <ButtonBackIcon onClick={() => history.push('/')} data-testid="button-back"><BackIconDark/></ButtonBackIcon>
            </ButtonWrapper>
        </GameDetailsCard>
    )
}
    
const GameDetailsCard = styled.section`
    margin: 0 20px;
`
const GameKeyInfos = styled.div`
    padding: 20px;
    display: grid;
    grid-template-rows: repeat(4, auto);
    grid-template-columns: 1fr 1fr;
    grid-gap: 10px;
    box-shadow: 0 0 10px var(--box-shadow-green);
    border-radius: 25px 25px 0 0;
    background-color: var(--text-light);
    font-size: 0.9rem;
`
const Date = styled.div`
    grid-column-start: 2;
    grid-row-start: 1;
    justify-self: end;
`
const Location = styled.div`
    grid-column: 1 / 3;
    grid-row-start: 2; 
    text-transform: uppercase;
`
const Players = styled.div`
    grid-column: 1 / 4;
    grid-row-start: 3;
`
const Winner = styled.div`
    grid-column-start: 1;
    grid-row-start: 4;
`
const Shots = styled.div`
    grid-column-start: 2;
    grid-row-start: 4;
`
const GameScores = styled.section`
    margin: 0 auto;
    padding: 20px;
    display: grid;
    grid-template-columns: 1fr 3fr;
    gap: 20px;
    background-color: var(--text-light);
`
const LegendHoles = styled.div`
    padding-right: 20px;
    display: grid;
    grid-template-rows: repeat(19, 35px);
    align-items: center;
    border-right: 1px solid var(--separator);

    span {
        width: 100%;
        text-align: center;
    }
`
const ScoresAllPlayers = styled.div`
    display: flex;
    overflow-y: auto;
    scrollbar-width: none;
    
    &::-webkit-scrollbar {
        display: none;
    }
`
const ScoresSinglePlayer = styled.div`
    margin: 0 2px;
    padding-left: 1px;
    display: grid;
    grid-template-rows: repeat(19, 35px);
    align-items: center;
    min-width: 80px;
`
const SingleScore = styled.div`
    display: grid;
    gap: 10px;
`
const ButtonWrapper = styled.div`
    padding: 5px 0;
    display: flex;
    justify-content: space-evenly;
    align-items: baseline;
    width: 100%;
    border-radius: 0 0 25px 25px;
    background: var(--text-light-transparent);
`
const ButtonBackIcon = styled.button`
    margin: 5px;
    padding: 0;
    display: flex;
    border: none;
    background: none;
`