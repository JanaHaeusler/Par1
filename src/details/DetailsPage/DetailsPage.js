import PropTypes from 'prop-types'
import GameDetails from '../GameDetails'

DetailsPage.propTypes = {
    targetProfile: PropTypes.object.isRequired,
    showOverviewPage: PropTypes.func.isRequired,
}

export default function DetailsPage({targetProfile, showOverviewPage}) {

    return (
        <GameDetails
        targetProfile={targetProfile}
        showOverviewPage={showOverviewPage}
        />
    )
}