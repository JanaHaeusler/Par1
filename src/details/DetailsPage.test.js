import { render } from '@testing-library/react'
import DetailsPage from './DetailsPage'

const testProps = {
    targetProfile: {
        location: 'City Park',
        date: '2020-12-24',
        winner: 'Sarah',
        shots: '28',
        playersString: 'Sarah, Hannes',
        playersArray: ['Sarah', 'Hannes'],
        scores: {
          Sarah: {
            hole1: '1',
            hole2: '2',
            hole3: '3',
            hole4: '4',
            hole5: '5',
            hole6: '6',
            hole7: '7',
            hole8: '1',
            hole9: '2',
            hole10: '3',
            hole11: '4',
            hole12: '5',
            hole13: '6',
            hole14: '7',
            hole15: '1',
            hole16: '2',
            hole17: '3',
            hole18: '4',
          },
          Hannes: {
            hole1: '7',
            hole2: '6',
            hole3: '5',
            hole4: '4',
            hole5: '3',
            hole6: '2',
            hole7: '1',
            hole8: '7',
            hole9: '6',
            hole10: '5',
            hole11: '4',
            hole12: '3',
            hole13: '2',
            hole14: '1',
            hole15: '7',
            hole16: '6',
            hole17: '5',
            hole18: '4',
          },
        },
        _id: '1',
      },
}

describe('DetailsPage', () => {
    it('renders snapshot correctly', () => {
        const { container } = render(<DetailsPage {...testProps}/>)
        expect(container.firstChild).toMatchSnapshot()
    })
})