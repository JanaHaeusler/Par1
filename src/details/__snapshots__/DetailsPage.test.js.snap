// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders snapshot correctly 1`] = `
.c13 {
  stroke: var(--primary);
}

.c0 {
  margin: 0 20px;
}

.c1 {
  border-radius: 25px 25px 0 0;
  background-color: var(--white);
  font-size: 1rem;
}

.c2 {
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: repeat(4,auto);
  grid-gap: 10px;
}

.c3 {
  grid-column-start: 2;
  grid-row-start: 1;
  justify-self: end;
}

.c4 {
  grid-column: 1 / 3;
  grid-row-start: 2;
  text-transform: uppercase;
}

.c5 {
  grid-column: 1 / 4;
  grid-row-start: 3;
}

.c6 {
  grid-column-start: 1;
  grid-row-start: 4;
}

.c7 {
  grid-column-start: 2;
  grid-row-start: 4;
}

.c8 {
  width: 90%;
  margin: 0 auto;
  padding: 10px 20px;
  display: grid;
  grid-template-columns: 1fr 3fr;
  gap: 20px;
}

.c9 {
  padding-right: 20px;
  display: grid;
  grid-template-rows: repeat(19,35px);
  border-right: var(--border-light);
}

.c9 span {
  text-align: center;
  font-weight: 600;
}

.c9 span:first-child {
  font-family: 'Montserrat',sans-serif;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-y: auto;
  -webkit-scrollbar-width: none;
  -moz-scrollbar-width: none;
  -ms-scrollbar-width: none;
  scrollbar-width: none;
}

.c10::-webkit-scrollbar {
  display: none;
}

.c11 {
  min-width: 80px;
  margin: 0 2px;
  display: grid;
  grid-template-rows: repeat(19,35px);
}

.c11 span:first-child {
  font-family: 'Montserrat',sans-serif;
  font-weight: 600;
}

.c12 {
  width: 100%;
  padding: 5px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  border: none;
  border-radius: 0 0 25px 25px;
  background: var(--white-transparent);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <section
      class="c2"
    >
      <div
        class="c3"
      >
        2020-12-24
      </div>
      <div
        class="c4"
      >
        City Park
      </div>
      <div
        class="c5"
      >
        <h4>
          Player(s)
        </h4>
        <span>
          Sarah, Hannes
        </span>
      </div>
      <div
        class="c6"
      >
        <h4>
          Winner(s)
        </h4>
        <span>
          Sarah
        </span>
      </div>
      <div
        class="c7"
      >
        <h4>
          Total Shots
        </h4>
        <span>
          28
        </span>
      </div>
    </section>
    <section
      class="c8"
    >
      <div
        class="c9"
      >
        <span>
          Holes
        </span>
        <span>
          1
        </span>
        <span>
          2
        </span>
        <span>
          3
        </span>
        <span>
          4
        </span>
        <span>
          5
        </span>
        <span>
          6
        </span>
        <span>
          7
        </span>
        <span>
          8
        </span>
        <span>
          9
        </span>
        <span>
          10
        </span>
        <span>
          11
        </span>
        <span>
          12
        </span>
        <span>
          13
        </span>
        <span>
          14
        </span>
        <span>
          15
        </span>
        <span>
          16
        </span>
        <span>
          17
        </span>
        <span>
          18
        </span>
      </div>
      <div
        class="c10"
      >
        <div
          class="c11"
        >
          <span>
            Sarah
          </span>
          <span>
            1
          </span>
          <span>
            2
          </span>
          <span>
            3
          </span>
          <span>
            4
          </span>
          <span>
            5
          </span>
          <span>
            6
          </span>
          <span>
            7
          </span>
          <span>
            1
          </span>
          <span>
            2
          </span>
          <span>
            3
          </span>
          <span>
            4
          </span>
          <span>
            5
          </span>
          <span>
            6
          </span>
          <span>
            7
          </span>
          <span>
            1
          </span>
          <span>
            2
          </span>
          <span>
            3
          </span>
          <span>
            4
          </span>
        </div>
        <div
          class="c11"
        >
          <span>
            Hannes
          </span>
          <span>
            7
          </span>
          <span>
            6
          </span>
          <span>
            5
          </span>
          <span>
            4
          </span>
          <span>
            3
          </span>
          <span>
            2
          </span>
          <span>
            1
          </span>
          <span>
            7
          </span>
          <span>
            6
          </span>
          <span>
            5
          </span>
          <span>
            4
          </span>
          <span>
            3
          </span>
          <span>
            2
          </span>
          <span>
            1
          </span>
          <span>
            7
          </span>
          <span>
            6
          </span>
          <span>
            5
          </span>
          <span>
            4
          </span>
        </div>
      </div>
    </section>
  </div>
  <button
    class="c12"
    data-testid="button-back"
  >
    <svg
      class="c13"
    >
      arrow-left.svg
    </svg>
  </button>
</div>
`;
